File: igraph/fullRun.php
<!DOCTYPE html>
<html>
<head>
	<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
</head>
<body>
<?php
include_once('../dbco.php');
include_once('../mainFunc.php');
include_once('../algo/func.php');

set_time_limit(4*3600);
$python = wherePython();
$limit = 1500;
if(isset($_GET['reset'])) {
	mysql_query("TRUNCATE TABLE `wg_cluster`");
	mysql_query("UPDATE wg_page SET cluster1=0, cluster2=0, cluster3=0, cluster4=0");
}

$clus = mysql_query("SELECT * FROM wg_cluster WHERE complete=0 ORDER BY id");
if($clust = mysql_fetch_array($clus)) {
	$level = $clust['level']; $cluster = $clust['id'];
}
else  {
	$any = mysql_query("SELECT * FROM wg_cluster");
	if(!$any = mysql_fetch_array($any)) {$level = 0; $cluster = 0;}
	else {echo "Program is done running.<br />";}
}

if(isset($level)) {
	emptyFolder('data');
	echo "Level: ".$level.". Cluster: ".$cluster.".<br />"; fl();
	
	include_once('1-buildGraph.php');
	if($level == 0) {
		emptyFolder('txt');
		buildSummaries($limit);
		echo "Built page summaries<br />"; fl();
	}

	createGraph($limit, $level, $cluster);
	echo 'Built graph for community detection<br />'; fl();

	$pyscript = '"'.getDocumentRoot().'/igraph/2-community.py"';
	$param1 = '"'.getDocumentRoot().'"';
	exec($python.' '.$pyscript." ".$param1, $output);
	echo 'Ran community detection<br />'; fl();

	$pyscript = '"'.getDocumentRoot().'/igraph/3-closeness.py"';
	exec($python.' '.$pyscript." ".$param1, $output);
	echo 'Scored communities with NLP; Generated community names<br />'; fl();

	$pyscript = '"'.getDocumentRoot().'/igraph/4-extrapolate.py"';
	exec($python.' '.$pyscript." ".$param1, $output);
	echo 'Extrapolated other nodes<br />'; fl();

	include_once('5-saveResults.php');
	saveResults($level, $cluster);
	echo "Saved results to database."; fl();
	echo 'All done.<br />';
	// echo '<script>window.location.reload();</script>';
}
function fl() {ob_flush(); flush();}
?>
</body>
</html>

File: igraph/fullRun.py
#!/usr/bin/python
import shutil

from dbco import *
from s1Build import createGraph
from s2Community import buildCommunity
from s3Nlp import useNLP
from s4QA import QA
from s5Extrapolate import extrapolate
from s6Save import saveResults

limit = 1500

if len(sys.argv) > 2 and sys.argv[2] == 'reset':
	shall = raw_input("Sure you want to reset DB? (y/N) ").lower() == 'y'
	if shall:
		cur.execute("TRUNCATE TABLE `wg_cluster`")
		cur.execute("UPDATE wg_page SET cluster1=0, cluster2=0, cluster3=0, cluster4=0, cluster5=0")

for i in range(0,20):
	count = cur.execute('SELECT id, level FROM wg_cluster WHERE complete=0 ORDER BY id LIMIT 1')
	level = -1; cluster = -1;
	if count > 0:
		rs = cur.fetchall()
		cluster = int(rs[0][0]); level = int(rs[0][1]);
	else:
		count = cur.execute('SELECT id, level FROM wg_cluster')
		if count < 1:
			level = 0; cluster = 0; # we have to build the first one.

	if level >= 0: # there's still something to run
		print "Level: ", level, ". Cluster:", cluster
		
		createGraph(limit, level, cluster)
		print "Built graph for community detection"

		buildCommunity(root)
		print "Ran community detection"

		useNLP(root)
		print "Reassign nodes with NLP + name communities"

		QA(root)
		print "Ran Q&A check on comunities"

		extrapolate(root)
		print "Extrapolated other nodes"

		saveResults(root, level, cluster)
		print "Saved results to database."
		print "All done.\n----------------------------"